### DISCUSSION TITLE
Can I split #[graphql_object] impl Query into multiple files?

### DISCUSSION THREAD

**Discussion by rimutaka** [graphql-rust/juniper#1045] [Q&A] [answered]
I have quite a few resolvers inside this impl block and would like to split it into multiple files
```rust
#[graphql_object(context=Ctx, scalar = RustScalarValue)]
impl Query {
    async fn resolver1<'db>(&self, context: &'db Ctx, ...) -> FieldResult<i32> { ... } 
    async fn resolver2<'db>(&self, context: &'db Ctx, ...) -> FieldResult<i32> { ... } 
    async fn resolver3<'db>(&self, context: &'db Ctx, ...) -> FieldResult<i32> { ... } 
// more resolvers ...
}
```

If I try to split it into multiple files it gives me this error
```
conflicting implementations of trait `juniper::macros::reflect::BaseType<stm_shared::graphql::RustScalarValue>` for type `graphql::Query`
conflicting implementation for `graphql::Query`
```
 as soon as I add `#[graphql_object]` to that other `impl Query` block. I think this is because the macro implements necessary traits multiple times. 

I wonder if there is a way around that.

**Comment by tyranron** [comment, marked as answer]
@rimutaka at the moment: no.

See more detailed discussion in https://github.com/graphql-rust/juniper/issues/646.
